packages/jellycommands/src/JellyCommands/commands/CommandManager.ts

```diff
@@ -24,19 +25,35 @@ export class CommandManager {
     }
 
     async respond(interaction: Interaction): Promise<void> {
-        if (!(interaction.isCommand() || interaction.isContextMenu())) return;
+        if (
+            !(
+                interaction.isCommand() ||
+                interaction.isContextMenu() ||
+                interaction.isAutocomplete()
+            )
+        )
+            return;
 
         const command = this.commands.get(interaction.commandId);
 
         // If command is not found return - if unknownCommand message send
         if (!command)
             return void (
+                !interaction.isAutocomplete() &&
                 this.client.joptions.messages?.unknownCommand &&
                 interaction.reply(this.client.joptions.messages.unknownCommand)
             );
 
         const options = command.options;
 
+        // If autocomplete interaction, run options.autocomplete
+        if (interaction.isAutocomplete())
+            /** @todo Duck you typescript */
+            return void (command as unknown as Command).autocomplete?.({
+                interaction,
+                client: this.client,
+            });
+
```

Comment:
How come you need to do this? It should be typed as Command not BaseCommand

Reply:
command is BaseCommand there because it can be either a `Command` (slash) or any of the other two, and it can't infer that an autocomplete interaction would only happen on a slash command.

(Also, the last part might not be true for modal interactions, not sure, I can't find anything about that)

Reply:
In the future might add a type option to basecommand but for now this is fine